src/Network.cpp

```diff
@@ -19,20 +19,38 @@ QNetworkReply* Network::get(QNetworkRequest request) {
     return nam_->get(request);
 }
 
-QNetworkReply* Network::downloadFile(QUrl url, QFile *dest) {
+QNetworkReply* Network::downloadFile(QUrl url, QFile *dest, QCryptographicHash::Algorithm algorithm, QByteArray hash) {
     QNetworkReply *reply = get(QNetworkRequest(url));
-    
+
     // Open in read/write so we can easily read the data when handling the
     // downloadComplete signal.
     if (!dest->open(QIODevice::ReadWrite)) {
         emit error(tr("Cannot open file for downloading."));
         return nullptr;
     }
+
+    // Optionally we have a hasher that hashes the download as it comes in and
+    // compares it against the provided hash when the download completes.
+    QCryptographicHash *hasher = nullptr;
```

Comment:
Why not use a smart ptr here, as opposed to a signal for freeing later?

Reply:
Because that's the sensible thing to do.

Reply:
Sorry, could you elaborate? Why not do something like `std::unique_ptr<QCryptographicHash> hasher(nullptr);`

Reply:
Oh you've already changed it.